great_expectations/datasource/fluent/sources.py

```diff
@@ -512,21 +517,14 @@ def update_datasource(
                 self._data_context.datasources.get(datasource_name), "id", None
             )
             if id_:
-                # if not a str `name_or_datasource` is a datasource and `id` can be directly attached
-                if name_or_datasource and not isinstance(name_or_datasource, str):
-                    name_or_datasource.id = id_
-                else:
-                    kwargs["id"] = id_
+                updated_datasource.id = id_
 
-            # local delete only, don't update the persisted store entry
-            self._data_context._delete_fluent_datasource(
-                datasource_name=datasource_name, _call_store=False
-            )
-            # Now that the input is validated and the old datasource is deleted we pass the
-            # original arguments to the add method (ie name and not datasource_name).
-            return self.create_add_crud_method(datasource_type)(
-                name_or_datasource, **kwargs
+            updated_datasource.test_connection()
```

Comment:
Why are we calling `test_connection()` here?

Reply:
See above comment: https://github.com/great-expectations/great_expectations/pull/8773#discussion_r1353428484